processing/src/main/java/io/druid/query/metadata/metadata/ListColumnIncluderator.java

```diff
@@ -64,12 +64,13 @@ public boolean include(String columnName)
     for (String column : this.columns) {
       final byte[] bytes = StringUtils.toUtf8(column);
       columns.add(bytes);
-      size += bytes.length;
+      size += bytes.length + 1;
     }
 
     final ByteBuffer bytes = ByteBuffer.allocate(size).put(LIST_CACHE_PREFIX);
     for (byte[] column : columns) {
       bytes.put(column);
+      bytes.put((byte) 0);
```

Comment:
Other places we use bytes that would render the byte array invalid UTF8. I think it was `0xFF`